core/src/main/scala/kafka/coordinator/transaction/TransactionMarkerChannelManager.scala

```diff
@@ -142,11 +141,13 @@ class TransactionMarkerChannelManager(config: KafkaConfig,
 
   def start(): Unit = {
     txnMarkerSendThread.start()
-    networkClient.wakeup()    // FIXME: is this really required?
   }
 
   def shutdown(): Unit = {
+    // wake up the thread in case it is blocked inside poll
+    networkClient.wakeup()
```

Comment:
We don't want txnMarkerSendThread to process any new requests after networkClient is waking up. So, a better shutdown sequence is probably:

txnMarkerSendThread.initiateShutdown()
networkClient.wakeup()
txnMarkerSendThread.awaitShutdown()